src/compiler/scala/tools/nsc/backend/jvm/BCodeHelpers.scala

```diff
@@ -876,7 +881,16 @@ abstract class BCodeHelpers extends BCodeIdiomatic {
       }
       debuglog(s"Potentially conflicting names for forwarders: $conflictingNames")
 
-      for (m <- moduleClass.info.membersBasedOnFlags(BCodeHelpers.ExcludedForwarderFlags, symtab.Flags.METHOD)) {
+      // Before erasure * to exclude bridge methods. Excluding them by flag doesn't work, because then
+      // the the method from the base class that the bridge overrides is included (scala/bug#10812).
+      // * using `exitingPickler` (not `enteringErasure`) because erasure enters bridges in traversal,
+      //   not in the InfoTransform, so it actually modifies the type from the previous phase.
+      //
+      // To keep binary compatibility with 2.12, we will still emit bridge methods, but with BRIDGE flag.
+      // Without it, Java 11 finds them to be ambiguous. (scala/bug#11061)
+      val membersAfterPickler = exitingPickler(moduleClass.info.membersBasedOnFlags(BCodeHelpers.ExcludedForwarderFlags, symtab.Flags.METHOD)).toList
+      val members = moduleClass.info.membersBasedOnFlags(BCodeHelpers.ExcludedForwarderFlags, symtab.Flags.METHOD)
+      for (m <- members) {
```

Comment:
could we, instead, get the method symbols in the same way as before, and just make sure we copy over the `bridge` flags to the forwareder methods? this would be the least risky change.

Reply:
@eed3si9n ah, that was because bridges were not included in the member search... so instead, the member search returned the method from the superclass (which the bridge would then override / implement), and this one doesn't have the bridge flag of course. Here's a fix: https://github.com/scala/scala/compare/2.12.x...lrytz:pr7035?expand=1. I think you can squash all these commits into one.

Reply:
Gotcha. Thanks for the hand holding :) A squashed commit is pushed with `[nomerge]`.